gbpservice/neutron/nsf/configurator/modules/firewall.py

```diff
@@ -0,0 +1,430 @@
+                msg = ("Failed to delete Firewall and status is "
+                       "changed to ERROR. %s." % str(err).capitalize())
+                LOG.error(msg)
+                # raise(err)
+            else:
+                if status == const.STATUS_ERROR:
+                    self.plugin_rpc.set_firewall_status(
+                        context, firewall['id'], status)
+                else:
+                    LOG.info("Firewall %r deleted of tenant: %r" % (
+                            firewall['id'], firewall['tenant_id']))
+                    self.plugin_rpc.firewall_deleted(
+                                        context, firewall['id'])
+
+        elif ev.id == 'update_firewall':
+            if not self._is_firewall_rule_exists(firewall):
+                return self.plugin_rpc.set_firewall_status(
+                    context, firewall['id'], const.STATUS_ACTIVE)
+            try:
+                status = self.method(None, None, firewall)
+            except Exception as err:
+                self.plugin_rpc.set_firewall_status(
+                            context, firewall['id'], 'ERROR')
+                msg = ("Failed to update Firewall and status is "
+                       "changed to ERROR. %s." % str(err).capitalize())
+                LOG.error(msg)
+            else:
+                self.plugin_rpc.set_firewall_status(
+                                context, firewall['id'], status)
+                msg = ("Updated Firewall and status set to %s" % status)
+                LOG.info(msg)
+        else:
+            raise Exception("Wrong call to Fwaas event handler.")
+
+
+def _create_rpc_agent(sc, topic, manager):
+    return RpcAgent(sc,
+                    host=cfg.CONF.host,
+                    topic,
+                    manager)
+
+
+def rpc_init(sc, conf):
+    fw_rpc_mgr = FwaasRpcReceiver(conf, sc)
+    fw_generic_rpc_mgr = FwGenericConfigRpcReceiver(conf, sc)
+
+    fw_agent = _create_rpc_agent(sc, const.FIREWALL_RPC_TOPIC, fw_rpc_mgr)
+    fw_generic_agent = _create_rpc_agent(
+                                    sc,
+                                    const.FIREWALL_GENERIC_CONFIG_RPC_TOPIC,
+                                    fw_generic_rpc_mgr)
+
+    sc.register_rpc_agents([fw_agent, fw_generic_agent])
+
+
+def events_init(sc, drivers):
+    evs = [
+        Event(id='CREATE_FIREWALL', handler=FwaasHandler(sc, drivers)),
+        Event(id='UPDATE_FIREWALL', handler=FwaasHandler(sc, drivers)),
+        Event(id='DELETE_FIREWALL', handler=FwaasHandler(sc, drivers)),
+
+        Event(id='CONFIGURE_INTERFACES', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='CLEAR_INTERFACES', handler=FwGenericConfigHandler(
+                                                                sc, drivers)),
+        Event(id='CONFIGURE_LICENSE', handler=FwGenericConfigHandler(
```

Comment:
Let us remove license related events and handlers from the code as this is not needed for Open Source release